python/lsst/ctrl/bps/parsl/sites/work_queue.py

```diff
@@ -0,0 +1,140 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from parsl.executors import WorkQueueExecutor
+from parsl.executors.base import ParslExecutor
+from parsl.launchers import SrunLauncher
+from parsl.providers import LocalProvider
+from parsl.providers.provider_base import ExecutionProvider
+
+from ..configuration import get_bps_config_value
+from ..site import SiteConfig
+
+if TYPE_CHECKING:
+    from ..job import ParslJob
+
+__all__ = ("WorkQueue", "SlurmWorkQueue")
+
+
+Kwargs = Dict[str, Any]
+
+
+class WorkQueue(SiteConfig):
+    """Configuraton for WorkQueueExecutor.
+
+    This must be subclassed, providing implementations for ``.get_executors``
+    and ``.select_executor``.  In ``.get_executors``, the site-specific
+    `ExecutionProvider` must be defined.
+
+    The following BPS configuration parameters are recognized, overriding the
+    defaults:
+
+    - ``port`` (`int`): The port used by the work_queue module. Default: `9000`.
+    - ``worker_options (`str`): Extra options to pass to work_queue workers.
+      a typical option sets the memory available per worker, e.g.,
+      `"--memory=90000"`, which sets the available memory to 90 GB. Default: `""`
+    - ``wq_max_retries`` (`int`): The number of retries that the work_queue
+      module will make when a task fails.  Set to `None` to have work_queue
+      retry forever; set to `1` to have retries managed by Parsl.  Default: 1
+    """
+
+    def __init__(self, *args, **kwargs):
+        # Have bps-defined resource requests for each job passed to work_queue.
+        kwargs["add_resources"] = True
+        super().__init__(*args, **kwargs)
+
+    def make_executor(
+        self,
+        label: str,
+        provider: ExecutionProvider,
+        *,
+        port: Optional[int] = 9000,
+        worker_options: Optional[str] = "",
+        wq_max_retries: Optional[int] = 1,
```

Comment:
Are these truly `Optional`, in the sense that `None` is an acceptable value? If so, `None` might be a better default value for at least some.

Reply:
Unfortunately, Parsl doesn't handle `None` as a default value.  For `port`, we could do what Parsl does, i.e., [this](https://github.com/Parsl/parsl/blob/master/parsl/executors/workqueue/executor.py#L212), but I found the lack of transparency of using that `WORK_QUEUE_DEFAULT_PORT` constant frustrating.  The other two defaults are exactly what Parsl has.

Reply:
I think @PaulPrice is saying that the type annotation is wrong and should be `int` if it can never be `None`.

Reply:
thanks for clarifying, Tim